src/main.py

```diff
@@ -13,19 +8,55 @@
 If an ARTag exists, the vehicle will become a follower vehicle.
 If an ARTag does not exist, the vehicle will become the leader vehicle.
 """
+
+
+from follower import Follower
+from leader import Leader
+from tagrec import TagRecognition
+
+import subprocess
+import time
+
+
+# This function calls the ./bt_toggle script with the specified command and address
+def bt_toggle(command, address):
+    subprocess.call(['./bt_toggle.sh', command, address])
+
+
 def main():
     tag = TagRecognition()
+    bt_address = '5C:BA:37:26:6D:9A'
 
     # If an ARTag is detected, there is a vehicle ahead of this vehicle so it will follow.
     # If an ARTag is not detected, no vehicle is present so this vehicle will be a leader.
     if tag.detect() is not None:
-        follower = Follower()
-        while True:
-            follower.follow()
+        bt_toggle('disconnect', bt_address)
+        is_leader = False
+        vehicle = Follower()    
     else:
-        leader = Leader()
-        while True:
-            leader.lead()
+        bt_toggle('connect', bt_address)
+        is_leader = True
+        vehicle = Leader()
+
+    # This loop makes vehicle move and if the vehicle is a 
+    # leader and sees an ARTag it will become a follower and
+    # disconnect from the bluetooth controller
+    start_time = time.time()
+    while True:
```

Comment:
While this whole block does what we want, it does it completely backwards. Why is it setting the timer if it sees the tag? All of this should be taken care of when we don't see a tag. it should start the timer when it doesn't see the tag and then build the rest of the algorithm around that.

Reply:
The problem is if I start the timer when I don't see a tag then I have to add another loop within this while loop since the timer would be inside of it, this way I felt like it was cleaner and still does exactly what we need. I did remove the start_time = time.time() line before the loop though as it was unnecessary.

Reply:
Sure we can go with that, but I think we need to have timer above the while loop still because say the vehicle never sees the tag in front of it, it will throw an error saying "undefined variable start_time" because it never gets initialized in the tag.detect() if statement.

Reply:
ok I can add that in there